language-support/ts/codegen/src/TsCodeGenMain.hs

```diff
@@ -233,25 +234,27 @@ daml2ts Daml2TsParams {..} = do
 
 -- Generate the .ts content for a single module.
 genModule :: Map.Map PackageId (Maybe PackageName, Package) ->
-     Scope -> PackageId -> Module -> Maybe (T.Text, [Dependency])
+     Scope -> PackageId -> Module -> Maybe (T.Text, Set.Set Dependency)
 genModule pkgMap (Scope scope) curPkgId mod
   | null serDefs =
     Nothing -- If no serializable types, nothing to do.
   | otherwise =
     let (defSers, refs) = unzip (map (genDataDef curPkgId mod tpls) serDefs)
-        imports = [ importDecl pkgMap modRef rootPath
-                  | modRef@(pkgRef, _) <- modRefs refs
-                  , let rootPath = pkgRootPath modName pkgRef ]
+        imports = Set.toList ((PRSelf, modName) `Set.delete` Set.unions refs)
+        (internalImports, externalImports) = splitImports imports
         defs = map biconcat defSers
-        modText = T.unlines $ intercalate [""] $ filter (not . null) $ modHeader : imports : defs
-        depends = [ Dependency $ pkgRefStr pkgMap pkgRef
-                  | (pkgRef, _) <- modRefs refs, pkgRef /= PRSelf ]
+        modText = T.unlines $ intercalate [""] $ filter (not . null) $
+          modHeader
+          : map (externalImportDecl pkgMap) externalImports
+          : map internalImportDecl internalImports
+          : defs
+        depends = Set.fromList [ Dependency $ pkgRefStr pkgMap pkgRef
+                  | (pkgRef, _) <- imports, pkgRef /= PRSelf ]
    in Just (modText, depends)
   where
     modName = moduleName mod
     tpls = moduleTemplates mod
     serDefs = defDataTypes mod
-    modRefs refs = Set.toList ((PRSelf, modName) `Set.delete` Set.unions refs)
     modHeader =
       [ "// Generated from " <> modPath (unModuleName modName) <> ".daml"
       , "/* eslint-disable @typescript-eslint/camelcase */"
```

Comment:
Do we still need this after the recent change?

Reply:
Do you mean the camelcase exception? We still produce names like `DA_Template_Internal` which might trigger a warning.

Reply:
I did. Ok!